crates/merkle-tree/src/class.rs

```diff
@@ -71,6 +80,162 @@ impl<'tx> ClassCommitmentTree<'tx> {
         let commitment = ClassCommitment(update.root_commitment);
         Ok((commitment, update))
     }
+
+    /// Generates a proof for a given `key`
+    pub fn get_proof(
+        tx: &'tx Transaction<'tx>,
+        block: BlockNumber,
+        class_hash: ClassHash,
+    ) -> anyhow::Result<Option<Vec<TrieNode>>> {
+        let root = tx
+            .class_root_index(block)
+            .context("Querying class root index")?;
+
+        let Some(root) = root else {
+            return Ok(None);
+        };
+
+        let storage = ClassTrieStorage {
+            tx,
+            block: Some(block),
+        };
+
+        let casm = tx
+            .casm_hash_at(block.into(), class_hash)
+            .context("Querying CASM hash")?;
+
+        let Some(casm) = casm else {
+            return Ok(None);
+        };
+
+        MerkleTree::<PoseidonHash, 251>::get_proof(root, &storage, casm.view_bits())
+    }
+}
+
+/// A [Patricia Merkle tree](MerkleTree) used to calculate commitments to
+/// Starknet's Sierra classes.
+///
+/// It maps a class's [SierraHash] to its [ClassCommitmentLeafHash]
+///
+/// Tree data is persisted by a sqlite table 'tree_class'.
+
+pub struct ClassStorageTree<'tx> {
+    tree: MerkleTree<PoseidonHash, 251>,
+    storage: ClassStorage<'tx>,
+}
```

Comment:
After looking at the Pathfinder implementation, I don't see the point of this structure. While the contract tree is made of one main tree that maps contract addresses to contract state, and then N subtrees that map the storage key and values for each contract, the class tree is unique and maps class hashes to class metadata. This is already implemented by `ClassCommitmentTree`.

Long story short, `ClassStorageTree` does not appear to be relevant and IMO can be deleted entirely.